int32 coming_to
bool synch
bool stop
bool repeat
float32[] distance
float32[] velocity
